File: CarbonaraAngular/src/app/model/booking.ts
import { CarClass } from "./carclass";
import { Station } from "./station";
import { User } from "./user";

export class Booking {

    constructor(
       
        public startTime: number,
        public endTime: number,
        public startStation: Station,
        public endStation: Station,
        public carclass: CarClass
    )
    {}
    public returnedTime?: number;
    public remarks?: string;
    public cancelled?: boolean;
    public returned?: boolean;
    public startKilometers?: number;
    public endKilometers?: number;
    public carLicenseNumber?: string;
    public billAmount?: number;
    public id?: number;
    public user?: User;

    
}

File: CarbonaraWebAPI/Model/DAO/Booking.cs
using System.ComponentModel.DataAnnotations;

namespace CarbonaraWebAPI.Model.DAO
{
    public class Booking
    {

        public Booking()
        {

        }
        public Booking( DateTime starttime, DateTime endtime, DateTime bookingtime, string remarks, bool cancelled, bool returned, User user, Carclass carclass, Station startstation, Station endstation, Car car, Bill? bill)
        {

            Starttime = starttime;
            Endtime = endtime;
            Bookingtime = bookingtime;
            Remarks = remarks;
            Cancelled = cancelled;
            Returned = returned;
            User = user;
            Carclass = carclass;
            Startstation = startstation;
            Endstation = endstation;
            Car = car;
            Bill = bill;
        }

        [Key]
        public int Id { get; set; }
        [Required]
        public DateTime Starttime { get; set; }
        [Required]
        public DateTime Endtime { get; set; }
        [Required]
        public DateTime Bookingtime { get; set; }
        [Required]
        public DateTime ReturnedTime { get; set; }
        public string Remarks { get; set; }
        [Required]
        public bool Cancelled { get; set; } = false;
        [Required]
        public bool Returned { get; set; } = false;
        public User? User { get; set; }
        [Required]
        public Carclass Carclass { get; set; }

        public Station? Startstation { get; set; }
        [Required]
        public Station Endstation { get; set; }

        public float StartKilometers { get; set; }
        public Bill? Bill { get; set; }
        public float EndKilometers { get; set; }
        public Car? Car { get; set; }

    }
}
